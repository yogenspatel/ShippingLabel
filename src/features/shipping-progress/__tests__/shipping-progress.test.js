/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import ShippingProgress from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Shipping Progress Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShippingProgress />, div);
  });
  it("has called snapshot with shipping progress component", () => {
    const tree = shallow(<ShippingProgress stepProgress={3} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("calculates progress percentage when step progress = 3", () => {
    const tree = shallow(<ShippingProgress stepProgress={3} />);
    expect(tree.find(".progress-bar").prop('style')).toEqual({"width": "60%"});
  });
  it("calculates progress percentage when step progress = 1", () => {
    const tree = shallow(<ShippingProgress stepProgress={1} />);
    expect(tree.find(".progress-bar").prop('style')).toEqual({"width": "10%"});
  });
});
