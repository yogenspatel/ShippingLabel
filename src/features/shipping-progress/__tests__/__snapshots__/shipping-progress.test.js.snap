// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shipping Progress Component renders correctly has called snapshot with shipping progress component 1`] = `
<div
  className="progress mt-3 mb-3"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "60%",
      }
    }
  />
</div>
`;
